app/backend/app.py

```diff
@@ -348,16 +345,16 @@ async def setup_clients():
     )
 
     if USE_GPT4V:
-        if vision_key is None:
-            raise ValueError("Vision key must be set (in Key Vault) to use the vision approach.")
+
+        token_provider = get_bearer_token_provider(azure_credential, "https://cognitiveservices.azure.com/.default")
```

Comment:
did we still want to support keys for folks that didn't want to use RBAC? (not sure why you would, so I think it's OK to drop this)

Reply:
I considered that, since it was already implemented, and since -accelerator is supporting both keys and RBAC. However, I think we agree that keyless auth is the future, and if we were to support keys for this API, we should support for all of them, which would significantly complicate code. We also know there are issues with using keyvault with azd, particularly in combination with CI/CD (due to the current user losing access). So we want to reduce use of key vault when possible.
So I decided to remove it. Existing users of GPT4V will be upgraded to managed identity when they re-provision.